.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,34 @@
+name: Links
+
+on:
+
+  push:
+    paths:
+      - '.github/workflows/check-links.yaml'
+      - '.lycheeignore'
+
+  repository_dispatch:
+
+  workflow_dispatch:
+
+  schedule:
+    - cron: "00 09 * * *"
```

Comment:
Unless there's a reason not to, let's switch to running this on every PR.

It looks like there's a cache to mitigate rate limiting problems: https://github.com/lycheeverse/lychee-action?tab=readme-ov-file#utilising-the-cache-feature if we need it

Reply:
Added running on every PR. But kept the cron to have at least a daily check in case remote URLs disappear. Also, created a cache with max age 1 d so that every external url is verified max once per day.

Reply:
Sorry I should have been more clear - I was requesting that the workflow be switched to only run on pull requests. But you do make a good point about the cron job.

If we keep the cron job, please:

 - Remove push triggers (we only need PRs + cron)
 - Remove the `repository_dispatch` permission
 - Update the permission of the workflow to be `contents: read` only at the workflow level: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token#defining-access-for-the-github_token-permissions
 - Update the pull_request trigger types to match the SDK: https://github.com/strands-agents/sdk-python/blob/main/.github/workflows/test-lint-pr.yml#L4